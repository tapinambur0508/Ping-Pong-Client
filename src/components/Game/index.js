import React from 'react';

import Ball from './Ball';
import Court from './Court';
import MyPaddle from './MyPaddle';
import EnemyPaddle from './EnemyPaddle';

class Game extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      width: window.innerWidth,
      height: window.innerHeight
    }
  }

  componentDidMount() {
    window.addEventListener('resize', () => {
      this.setState({
        width: window.innerWidth,
        height: window.innerHeight
      });
    });
  }

  render() {
    return (
        <Court
          width={this.state.width}
          height={this.state.height}>
          <MyPaddle
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
          <Ball
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
          <EnemyPaddle
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
        </Court>
    );
  }
}

export default Game;
