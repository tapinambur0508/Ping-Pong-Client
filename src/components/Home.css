#home .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#home .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#home .wrapper nav {
  margin: 10px 0;
}

#home .wrapper nav a.btn {
  border-radius: 10px;
}