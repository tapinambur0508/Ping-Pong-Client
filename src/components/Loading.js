import React from 'react';

export default ({ error }) => {
  if (error) {
    return <div>Error</div>
  } else {
    return <div>Loading...</div>;
  }
}
