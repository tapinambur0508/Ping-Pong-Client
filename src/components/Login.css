@import url('https://fonts.googleapis.com/css?family=Poppins');
#login .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#login .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#login .wrapper button.kep-login-facebook {
  margin: 10px 0;
  border-radius: 50px;
  text-indent: 10px;
  border: none;
  width: 100%;
}