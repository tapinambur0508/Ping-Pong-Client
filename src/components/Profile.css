#profile .card {
  max-width: 350px;
  margin: 50px auto;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#profile .card .unit-stats {
  margin: 0;
  color: #ffffff;
  font-weight: 700;
  background: #eda54e;
}

#profile .card .unit-stats .item {
  padding: 1.2rem 1rem;
}

#profile .card .unit-stats .item:nth-child(2) {
  border-left: 1px solid #bc7c2f;
  border-right: 1px solid #bc7c2f;
}

#profile .card .unit-stats .item .stat {
  font-size: 1.5rem;
  margin-bottom: 8px;
}

#profile .card .unit-stats .item .stat-value {
  text-transform: uppercase;
  font-weight: 400;
  font-size: 0.8rem;
}