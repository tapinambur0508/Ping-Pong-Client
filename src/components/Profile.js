import React from 'react';
import axios from 'axios';

class Profile extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      user: {}
    }
  }

  componentDidMount() {
    axios.get('')
      .then(({ data }) => {

      })
      .catch(err => console.log(err));
  }

  render() {
    return (
      <section>
        <div className="container">
          <div className="card mx-auto my-3" style={{ width: "18rem" }}>
            <img 
              className="card-img-top" 
              src="images/default-avatar.jpg" alt="John Doe" />
            <div className="card-body">

            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default Profile;
