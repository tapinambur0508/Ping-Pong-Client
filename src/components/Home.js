import React from 'react';
import NavLink from 'react-router-dom/NavLink';

import './Home.css';

export default () => (
  <section id="home">
    <div className="container">
      <div className="wrapper">
        <h1 className="heading">Ping-Pong</h1>
        <nav>
          <NavLink to="/start-game" className="btn btn-success btn-block">
            <i className="fas fa-play mr-2"></i>Start Game
            </NavLink>
          <NavLink to="/profile" className="btn btn-info btn-block">
            <i class="far fa-user-circle mr-2"></i>Profile
          </NavLink>
        </nav>
      </div>
    </div>
  </section>
);
