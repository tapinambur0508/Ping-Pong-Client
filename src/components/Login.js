import React from 'react';
import FacebookLogin from 'react-facebook-login';
import withRouter from 'react-router-dom/withRouter';
import axios from 'axios';

import AuthService from '../services/auth';

import config from '../config.json';

import './Login.css';

const Login = props => {
  const facebookResponse = response => {
    axios({
      method: 'POST',
      url: 'https://ping-pong-main-server.herokuapp.com/api/oauth/facebook',
      data: {
        access_token: response.accessToken
      },
      headers: {
        'Content-Type': 'application/json'
      }
    })
      .then(({ data }) => {
        const authService = new AuthService();

        authService.token = data['access_token'];
        props.history.push('/');
      })
      .catch(err => console.log(err));
  };

  return (
    <section id="login">
      <div className="container">
        <div className="wrapper">
          <h1 className="heading">Ping-Pong</h1>
          <FacebookLogin
            appId={config.FACEBOOK_APP_ID}
            autoLoad={false}
            fields="name,email,picture"
            textButton="Continue with Facebook"
            callback={facebookResponse} />
        </div>
      </div>
    </section>
  );
}

export default withRouter(Login);
