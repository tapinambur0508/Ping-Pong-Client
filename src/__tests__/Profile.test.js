import React from 'react';
import { Profile } from '../components/Profile';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

describe('Profile', () => {
  it('should be defined', () => {
    expect(Profile).toBeDefined();
  });

  it('Profile Component matches the snapshot', () => {
    const context = {
      props: {}
    };
    const props = { 
      history: {
        push() {
  
        }
      }  
    };
    const tree = renderer.create(
      <StaticRouter context={context}>
        <Profile {...props}/>
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('initial state of the user object', () => {
    const props = { 
      history: {
        push() {
  
        }
      }  
    };
    const wrapper = shallow(<Profile {...props}/>);
    expect(wrapper.state().user).toEqual({
      account: {
        username: '',
        level: '',
        experience: '',
        coins: ''
      },
      facebook: {
        id: '',
        email: '',
        name: {
          familyName: '',
          givenName: '',
          middleName: ''
        }
      }
    });
  });

  it('Check state of the user object after logout', () => {
    const fakeUser = {
      account: {
        username: 'username',
        level: '1',
        experience: '0',
        coins: '10'
      },
      facebook: {
        id: '1111',
        email: 'email',
        name: {
          familyName: 'fn',
          givenName: 'gn',
          middleName: 'mn'
        }
      }
    };
    const props = { 
      history: {
        push() {
  
        }
      }  
    };
    const wrapper = shallow(<Profile {...props}/>);
    wrapper.setState({ user: fakeUser });
    expect(wrapper.state()).toEqual({ user: fakeUser });
    wrapper.find('.btn .btn-danger').simulate('click');
    expect(wrapper.state().user).toEqual({
      account: {
        username: '',
        level: '',
        experience: '',
        coins: ''
      },
      facebook: {
        id: '',
        email: '',
        name: {
          familyName: '',
          givenName: '',
          middleName: ''
        }
      }
    });
  });
})
