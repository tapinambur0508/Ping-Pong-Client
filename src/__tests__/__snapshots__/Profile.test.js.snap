// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile Component matches the snapshot 1`] = `
<section
  id="profile"
>
  <div
    className="container"
  >
    <div
      className="card text-center"
    >
      <div
        className="card-header"
      >
        <h2>
          Profile
        </h2>
      </div>
      <div
        className="card-body"
      >
        <h2
          className="card-title"
        >
          
          Â 
          
        </h2>
        <h5
          className="card-subtitle text-muted"
        >
          
        </h5>
      </div>
      <div
        className="row unit-stats"
      >
        <div
          className="col-4 item"
        >
          <div
            className="stat"
          >
            
          </div>
          <div
            className="stat-value"
          >
            Level
          </div>
        </div>
        <div
          className="col-4 item"
        >
          <div
            className="stat"
          >
            
          </div>
          <div
            className="stat-value"
          >
            Experience
          </div>
        </div>
        <div
          className="col-4 item"
        >
          <div
            className="stat"
          >
            
          </div>
          <div
            className="stat-value"
          >
            Coins
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <button
          className="btn btn-danger"
          onClick={[Function]}
        >
          <i
            className="fas fa-sign-out-alt mr-1"
          />
          Logout
        </button>
      </div>
      <div
        className="card-footer"
      >
        <a
          aria-current="page"
          className="btn btn-dark active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="fas fa-arrow-left mr-1"
          />
          Back
        </a>
      </div>
    </div>
  </div>
</section>
`;
